src/System.Net.HttpListener/src/System/Net/Managed/HttpListenerRequest.Managed.cs

```diff
@@ -182,9 +182,7 @@ internal void FinishInitialization()
             if (colon >= 0)
                 host = host.Substring(0, colon);
 
-            string base_uri = String.Format("{0}://{1}:{2}",
-                                (IsSecureConnection) ? "https" : "http",
-                                host, LocalEndPoint.Port);
+            string base_uri = $"{RequestScheme}://{host}:{LocalEndPoint.Port}";
```

Comment:
I thought it was equivilent to string.Format. Sad that this isn't a zero cost abstraction, but I'm sure the c# team have their reasons